customQueryGenerator.js

```diff
@@ -0,0 +1,170 @@
+		for (var i = 0; i < itemList.length; i++){
+			var newEntry = stringCheckForItem(itemList[i],
+			leftSideEquivalencyTest);
+			selectedItemStrings.push(newEntry);
+		}		
+		outputString += selectedItemStrings.join(' || ');
+		outputString += ')';
+		// set propertiesList variable for use elsewhere
+		propertiesList[dropdownMenu.stringName] = outputString;
+	}
+	// nothing was selected therefore set to null
+	else {
+		propertiesList[dropdownMenu.stringName] = null;
+	}
+}
+
+// Function to check an item input against select properties. If it's numerical
+// 		or undefined, put into string without quotations else add quotations
+// Inputs: selectedItem, String of selected item from dropdown menu
+//		leftSideEquivalencyTest, String similar to:
+//			'event.properties["App Name"] === '
+// Output: 'event.properties["App Name"] === "Robots"'
+function stringCheckForItem(selectedItem, leftSideEquivalencyTest){
+	var outputString = '';
+	// Due to test cases having numerical values, have created possibility
+	// but may become invalid if new options that are non-numerical and of
+	// length 1 come into play
+	// NOTE: isNaN function doesn't work when checking numbers in a string 
+	// so checking by length
+	if (isNaN(selectedItem) && selectedItem !== "undefined"){
+		outputString = leftSideEquivalencyTest + '"' + selectedItem + '"';
+	}
+	// A numerical input or undefined value, do not add quotations
+	else{
+		outputString = leftSideEquivalencyTest + selectedItem;
+	}
+	return outputString;
+}
+
+// Function to build the script for use in custom_query commands
+// Inputs: propertySelector, the string variant of the field 
+//		name to be groupedBy,
+//		selector, a variable determining if it's retrieving menu 
+//		data or usable data
+// 		selector == 0, retrieve menu data
+// 		selector == 1, retrieve table data
+function buildCustomQueryFunctionScript(propertySelector, selector){
+	var startFunction = '\
+		function main() { \
+			return Events({from_date: params.from_date,\
+				to_date: params.to_date,\
+				event_selectors: [{event: params.event}]})';
+	// filter by event
+	var eventFilter = '.filter(function(event) { return (event.name === params.event ) })';
+	// filter by properties (Platform, App Name, and $model)
+	var propertyFilters = '';
+	if (propertiesList.platformString){
+		propertyFilters += '.filter(function(event) { return (' + propertiesList.platformString +' ) })';
+	}
+	if (propertiesList.appNameString){
+		propertyFilters += '.filter(function(event) { return (' + propertiesList.appNameString +' ) })';
+	}
+	
+	if (propertiesList.deviceModelString){
+		propertyFilters += '.filter(function(event) { return (' + propertiesList.deviceModelString +' ) })';
+	}
+	var groupByString = '';
+	var reducer = '';
+	var map = '';
+	switch (selector) {
+		// Get data for dropdown menu options, returned as array of names
+		case 0:
+			groupByString = '.groupBy(["properties.'+ propertySelector +'"], mixpanel.reducer.count())';

```

Comment:
I don't see how. It's like in the example.js file in this set. The function is an actual function. It's morphed into a string using a prototype.toString function which converts the function as it is into a string. If I used a stringreplace on it after morphing it, is there a point in doing that at all since you would have the same result with a buttload of string replaces like this anyways? Maybe what you want is a commented portion of what the actual output should look like?